src/transports/smart_protocol.c

```diff
@@ -751,9 +751,10 @@ static int add_push_report_sideband_pkt(git_push *push, git_pkt_data *data_pkt,
 				git_buf_put(data_pkt_buf, line, line_len);
 			error = 0;
 			goto done;
-		}
-		else if (error < 0)
+		} else if (error) {
```

Comment:
When would we get a positive `error` here?  It's not immediately clear to me that `git_pkt_parse_line` would return a positive value and we would then want to fail because of it.

I worry a bit that static analysis has detected that this is possible, but (at a quick glance) it's not clear to me when this would happen, so it's not clear to me that it should be an error.  :/